python/graphstorm/dataloading/dataloading.py

```diff
@@ -917,3 +924,175 @@ def __iter__(self):
 
     def __next__(self):
         return self.dataloader.__next__(), self.unlabeled_dataloader.__next__()
+
+####################### Distillation #############################
+
+class DataManager:
+    r"""Data Manager. Combines a file sampler and a dataloader generator,
+    and streamingly provides an iterable over a set of files.
+
+    Parameters:
+    ----------
+        dataloader_generator : DataloaderGenerator: 
+            A dataloader generator
+            Generates dataloader based on given a file path.
+        dataset_path str : 
+            Path to the data files.
+        shuffle : bool
+            Set to ``True`` to have the files reshuffled at every epoch (default: ``False``).
+        infinite : bool
+            Set to ``True`` to have the files sampled infinitely (default: ``False``).
+        local_rank : int
+            Local rank for the trainer (default: ``-1``).
+        is_train : bool
+            Set to ``True`` if the provider is for training set (default: ``True``).
+    """
+
+    def __init__(
+        self,
+        dataloader_generator,
+        dataset_path,
+        shuffle=False,
+        infinite=False,
+        local_rank=-1,
+        world_size=1,
+        is_train=True,
+    ):
+        super().__init__()
+        # TODO (HZ): implement prefetch to put files into a queue
+        self.is_train = is_train
+        if dataset_path is not None:
+            if not isinstance(dataset_path, str):
+                raise TypeError(
+                    "dataset_path should be a str, but got "
+                    "dataset_path={}".format(dataset_path)
```

Comment:
Please also report the type of dataset_path.

Reply:
addressed by new commit.